src/opensearch.ts

```diff
@@ -0,0 +1,113 @@
+import { extractSource } from './extract'
+import { ApplicationError, StatusCodes } from './http'
+
+enum OSIndex {
+    Snapshots = 'snapshots_kris_1'
+}
+
+export const getClient = (): Client => {
+    return new Client({ node: 'http://localhost:9501' })
+}
+
+export const getNestedObject = (
+    path: string,
+    term: string,
+    value: string,
+    includes: string[]
+) => {
+    return {
+        path: path,
+        query: {
+            bool: {
+                filter: {
+                    term: {
+                        [path + '.' + term]: {
+                            value
+                        }
+                    }
+                }
+            }
+        },
+        ignore_unmapped: false,
+        inner_hits: {
+            ignore_unmapped: false,
+            from: 0,
+            size: 7,
+            version: false,
+            seq_no_primary_term: false,
+            explain: false,
+            track_scores: false,
+            _source: {
+                includes: includes.map(name => path + '.' + name)
+            }
+        }
+    }
+}
+
+export const getBalanceByAddress = (es: Client) => (address: string, ordinal: string = 'latest'): TaskEither<ApplicationError, any> => {
+
+    const innerPath = 'info.balances'
+    const nested = getNestedObject(innerPath, 'address', address, ['balance'])
+    const osSearch = getByFieldQuery(OSIndex.Snapshots, 'ordinal', ordinal, ['ordinal'], nested)(es)
+
+    return findOne(innerPath)(osSearch)
+}
+
+const getByFieldQuery = (
+    index: string,
+    field: string,
+    value: string,
+    includes: string[] = [],
+    nestedBody: any,
+    size: number | null = 1
+) => (es: Client) =>
+        es.search({
+            index,
+            body: {
+                size,
+                query: {
+                    bool: {
+                        filter: [
+                            {
+                                term: {
+                                    [field]: {
+                                        value: value
+                                    }
+                                }
+                            },
+                            { nested: nestedBody }
+                        ]
+                    }
+                },
+                _source: {
+                    includes: includes
+                }
+            }
+        })
+
+const concatResults = (res1: any[]) => (res2: any[]) => Object.assign(res1[0], res2[0])
+
+const findOne = (innerPath: string) => (search: TransportRequestPromise<ApiResponse>) => pipe(
+    tryCatch<ApplicationError, any>(
+        () => search.then(r =>
+            r.body.found
+                ? [r.body]
+                : r.body
+        ),
+        err => new ApplicationError(
+            'ElasticSearch error',
+            [err as string],
+            StatusCodes.SERVER_ERROR
+        )
+    ),
+    chain(body => pipe(
+        taskEither.of(concatResults),
```

Comment:
I think it can be
```typescript
taskEither.of(a => b => [...a, ...b])
```

Reply:
It is TS @marcinwadon  so `taskEither.of((a, b) => [...a, ...b])`

no implicit currying available there :D